src/system/SystemLayerImplSelect.cpp

```diff
@@ -170,6 +170,10 @@ void LayerImplSelect::CancelTimer(TimerCompleteCallback onComplete, void * appSt
     VerifyOrReturn(mLayerState.IsInitialized());
 
     TimerList::Node * timer = mTimerList.Remove(onComplete, appState);
+    if (timer == nullptr)
```

Comment:
Please document this very very subtle nullptr check here that removes from a different pool. Arguably it's one of the biggest parts of this PR

Reply:
Will do.